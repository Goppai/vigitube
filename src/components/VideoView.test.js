import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { InnerView as VideoView } from './'

describe('<VideoView/>', () => {
  const dummyProps = {
    selectedVideo: {
      id: {
        videoId: 'test-id'
      },
      snippet: {
        title: 'Test-title',
        description: 'Test-description'
      }
    },
    size: {
      width: 640
    }
  }
  const wrapper = shallow(<VideoView {...dummyProps}/>)

  it('renders', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders youtube iframe', () => {
    expect(wrapper.find('iframe').length).toBe(1)
  })

  it('renders video details', () => {
    expect(wrapper.find('.video-details').length).toBe(1)
  })
})
