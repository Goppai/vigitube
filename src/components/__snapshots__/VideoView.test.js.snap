// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoView/> renders 1`] = `
<Col
  className="video-view"
  lg={17}
  xs={24}
>
  <iframe
    allowFullScreen={true}
    className="width-100"
    frameBorder={0}
    src="https://www.youtube.com/embed/test-id?rel=0"
    style={
      Object {
        "height": 360,
      }
    }
    title="video-player"
  />
  <Card
    className="video-details"
    title="Test-title"
  >
    <Component
      description="Test-description"
    />
  </Card>
</Col>
`;
